### PR TITLE
Dedicated build instance

### PR DESCRIPTION
**Pull request by ddollar**
This change creates a new Rack parameter called `BuildInstance`. Set this parameter to an instance type to create a dedicated build instance (AutoScalingGroup of size 1).

`convox rack params set BuildInstance=c4.large`

### CODE REVIEW

File: provider/aws/dist/rack.json [hunk 1]
```diff
@@ -1321,6 +1329,128 @@
...
+            "  - mkdir -p /etc/convox\n",
+            "  - echo -e '/var/log/docker {\\n  rotate 7\\n  daily\\n  nocompress\\n  copytruncate\\n}' >> /etc/logrotate.d/docker\n",
+            "  - export INSTANCE_ID=$(curl -s http://169.254.169.254/latest/meta-data/instance-id)\n",
+            "  - /opt/aws/bin/cfn-signal --stack ", { "Ref": "AWS::StackName" }, " --region ", { "Ref":"AWS::Region" }, " --resource BuildInstances\n"
+          ] ] }
+        }
+      }
+    },
+    "BuildInstances": {
+      "DependsOn": [ "AvailabilityZones", "Subnet0", "Subnet1" ],
```
Comment (mattmanning) [inline provider/aws/dist/rack.json]: Do we need something conditional for private subnets in private racks?

Comment (ddollar) [inline provider/aws/dist/rack.json]: It automatically happens by being referenced if the `Private` condition is met.

**Comment by ddollar** [comment]
This should ideally always be 1 instance for best caching. If you need more space use a bigger instance.

**Comment by mwarkentin** [comment]
Agreed, I think getting the caching benefits will be huge, and you can use a beefy instance type if needed. Could the instance potentially leverage an EBS volume to persist the cache even if the instance is terminated, etc.?

I can imagine that we'll be configuring nightly scale downs for this build "cluster" (maybe even in prod?).. could be nice if that didn't blow away the cache.

**Comment by ddollar** [comment]
[dedicated build instance](https://trello.com/c/JpqmYNZn/1867-dedicated-build-instance)

File: provider/aws/dist/rack.json [hunk 2]
```diff
@@ -1935,7 +2066,7 @@
               "AWS_REGION": { "Ref": "AWS::Region" },
               "AWS_ACCESS": { "Ref": "KernelAccess" },
               "AWS_SECRET": { "Fn::GetAtt": [ "KernelAccess", "SecretAccessKey" ] },
-              "CLIENT_ID": { "Ref": "ClientId" },
```
Comment (MiguelMoll) [inline provider/aws/dist/rack.json]: Was this removed on purpose? This was needed for build notifications.

**Comment by MiguelMoll** [comment]
Needed these permissions while testing:

KernelUserGeneralPolicy

- autoscaling:CreateAutoScalingGroup
- ecs:CreateCluster

KernelUserPolicy 

```
        {
            "Action": "ecs:DeleteCluster",
            "Resource": [
                "arn:aws:ecs:*:*:cluster/{ "Ref": "AWS::StackName" }*"
            ],
            "Effect": "Allow",
            "Sid": "LimitedECSAccess"
        }
```

This one should be locked down
```
autoscaling:DeleteAutoScalingGroup
```

**Pull request closed without merging**